nx/lib/nx.ex

```diff
@@ -6149,8 +6149,33 @@ defmodule Nx do
     rtol = opts[:rtol]
     atol = opts[:atol]
 
-    # TODO: deal with non_finite entries by adding is_infinity and is_nan
-    all(less_equal(Nx.abs(subtract(a, b)), add(atol, multiply(rtol, Nx.abs(b)))))
+    a = to_tensor(a)
+    b = to_tensor(b)
+
+    nan_a = is_nan(a)
+    nan_b = is_nan(b)
+    nan_selector = logical_or(nan_a, nan_b)
+    nan_entries = logical_and(nan_a, nan_b)
+
+    inf_a = is_infinity(a)
+    inf_b = is_infinity(b)
+    inf_selector = logical_or(inf_a, inf_b)
+
+    inf_entries = logical_and(inf_selector, equal(a, b))
+
+    non_finite_selector = logical_or(inf_selector, nan_selector)
+
+    finite_entries =
+      select(
+        non_finite_selector,
+        non_finite_selector,
```

Comment:
I reckon `select` isn't casting something properly. Previously I had `select(non_finite_selector, 1, ...)` and that raised some errors